File: game-service/src/main/java/io/github/xpakx/habitgame/battle/dto/BattleResponse.java
package io.github.xpakx.habitgame.battle.dto;

import io.github.xpakx.habitgame.battle.BattleObjective;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class BattleResponse {
    private Long battleId;
    private Integer width;
    private Integer height;
    private boolean finished;
    private boolean started;
    private BattleObjective objective;
    private Integer turn;
}

File: frontend/src/app/battle/dto/battle-response.ts
import { BattleShip } from "./battle-ship";

export interface BattleResponse {
    battleId: number;
    width: number;
    height: number;
    finished: boolean;
    started: boolean;
    objective: String;
    turn: number;
    ships: BattleShip[];
    enemyShips: BattleShip[];
}